WordPress/gutenberg
# Move template areas to editor package

Follow up to https://github.com/WordPress/gutenberg/pull/59870.
The PostCardPanel panel component only accepted children to pass template areas.
At https://github.com/WordPress/gutenberg/pull/59870#discussion_r1527153908 @youknowriad suggested removing the children prop and using template areas directly on the component.
This PR does that change.

To do that change the following sub-changes were done.

- TemplateAreas component was moved to the editor package.
- The edit-site selector getCurrentTemplateTemplateParts was also moved to the editor package, making the existing edit-site version call the editor version for back-compatibility purposes.


## Testing Instructions
Open a template and verify the template areas still appear on the template inspector without any noticeable change when compared to the trunk.
